app/controllers/posts_controller.rb

```diff
@@ -27,9 +27,26 @@ def update
       render 'edit'
     end
   end
-
+# # search
+#   def search
+#     if query.present?
+#       Post.index
+#     else
+#       scoped
+#     end
+#   end
+
+#   def search
+#     Post.search params[:query]
+#   end
+  
   def index
-    @posts = Post.published.order('views_count DESC , created_at DESC')
+    @posts = Post.all
+    if params[:query].present?
+      @posts = Post.search(params[:query]).order('views_count DESC , created_at DESC')
+    else
+      @posts = Post.published.order('views_count DESC , created_at DESC')
```

Comment:
@babiracka There is a small duplication: `.order('views_count DESC , created_at DESC')`.

Imagine you add more conditions. Or more methods need to be specified in query. Are you going to copy–paste it to every case?

You could put that call outside the if statements instead :)